src/v/rpc/connection_cache.h

```diff
@@ -67,29 +93,14 @@ class connection_cache final
       model::node_id node_id,
       clock_type::time_point connection_timeout,
       Func&& f) {
-        using ret_t = result_wrap_t<std::invoke_result_t<Func, Protocol>>;
         auto shard = rpc::connection_cache::shard_for(self, src_shard, node_id);
 
         return container().invoke_on(
           shard,
           [node_id, f = std::forward<Func>(f), connection_timeout](
             rpc::connection_cache& cache) mutable {
-              if (!cache.contains(node_id)) {
```

Comment:
> invoke_on already checks if the target/source shards are the same and runs the function immediately
True, but that may not always be the case, since the connection cache will assign a shard for the connection via `rpc::connection_cache::shard_for`. The intent here is to have the connection on the same shard as the `node_status_backend`.

Reply:
> The intent here is to have the connection on the same shard as the node_status_backend.

i'm not sure i understand why this would be important as opposed to allowing the connection cache to distributed connections across the cores as usual.

Reply:
To wrap up this discussion: I ended up distributing the connections to peers across shards. See [this](https://github.com/redpanda-data/redpanda/compare/037cff2f8cb290994e8ab34111457a1332fb45f4..8479bb7f809ea2f5090c5b053ef678b048f45e04) force push for details.